download-basket/src/main/java/downloadbasket/helpers/SendDownloadDetailsToEmailThread.java

```diff
@@ -0,0 +1,289 @@
+
+						String folderName = zdd.getLayerName() + "_" + index + "/";
+						out.putNextEntry(new ZipEntry(folderName));
+
+						in = new ZipInputStream(new FileInputStream(strTempFile));
+						ZipEntry ze = in.getNextEntry();
+						while (ze != null) {
+							String fileName = ze.getName();
+							out.putNextEntry(new ZipEntry(folderName + fileName));
+							int len;
+							while ((len = in.read(buffer)) > 0) {
+								out.write(buffer, 0, len);
+							}
+							ze = in.getNextEntry();
+						}
+
+						out.closeEntry();
+						in.close();
+						deleteFile(strTempFile);
+
+					} catch (Exception ex) {
+						LOGGER.error("Cannot  parse JSON download details", ex);
+					} finally {
+						if (in != null)
+							in.close();
+
+					}
+				}
+
+			} catch (FileNotFoundException fe) {
+
+			} catch (Exception ex) {
+				ex.printStackTrace();
+			} finally {
+				if (out != null) {
+					try {
+						out.close();
+					} catch (Exception ex) {
+						LOGGER.error("Cannot close output", ex);
+					}
+				}
+			}
+			sendZipFile(strZipFileName);
+		} catch (Exception ex) {
+			LOGGER.error("Cannot download shape zip.", ex);
+		}
+	}
+
+	/**
+	 * 
+	 * Sends the zip file to current user's email address.
+	 * 
+	 * @param strZipFileName
+	 *            zip file name
+	 */
+	public void sendZipFile(final String strZipFileName) {
+
+		try {
+			HtmlEmail email = new HtmlEmail();
+
+			int smtpPort = Integer.parseInt(PropertyUtil.getNecessary("oskari.wfs.download.smtp.port"));
+			email.setSmtpPort(smtpPort);
+			email.setHostName(PropertyUtil.getNecessary("oskari.wfs.download.smtp.host"));
+			email.setFrom(PropertyUtil.getNecessary("oskari.wfs.download.email.from"));
+			email.setSubject(PropertyUtil.getNecessary("oskari.wfs.download.email.subject"));
+			email.setCharset("UTF-8");
+
+			StringBuilder htmlHeader = new StringBuilder();
+			StringBuilder htmlMsg = new StringBuilder();
+			StringBuilder htmlFooter = new StringBuilder();
+
+			StringBuilder txtHeader = new StringBuilder();
+			StringBuilder txtMsg = new StringBuilder();
+			StringBuilder txtFooter = new StringBuilder();
+
+			htmlHeader.append(PropertyUtil.getNecessary("oskari.wfs.download.email.header"));
+			txtHeader.append(PropertyUtil.getNecessary("oskari.wfs.download.email.header"));
+
+			htmlHeader.append("<br/><br/>");
+			txtHeader.append("\n\n");
+			htmlMsg.append(PropertyUtil.getNecessary("oskari.wfs.download.email.message"));
+			txtMsg.append(PropertyUtil.getNecessary("oskari.wfs.download.email.message"));
+
+			htmlMsg.append("<br/>");
+			txtMsg.append("\n");
+
+			String url = PropertyUtil.getNecessary("oskari.wfs.download.link.url.prefix") + strZipFileName;
+			htmlMsg.append("<a href=\"" + url + "\">" + url + "</a>");
+			txtMsg.append(url);
+
+			htmlFooter.append("<br/><br/>");
+			txtFooter.append("\n\n");
+			String f = PropertyUtil.get("oskari.wfs.download.email.footer", "");
+			String ff = f.replaceAll("\\{RIVINVAIHTO\\}", "\n");
```

Comment:
RIVINVAIHTO -> LINEBREAK if you must do it this way.

Reply:
Also you could extract at least "createMsg()" to another method.

Reply:
I think you can replace \\{RIVINVAIHTO\\} with \\n in the property values and remove this hack. 

This email message is constructed by values in the oskari-ext.properties that is not really localizable. Another message in DownloadServices class uses the spring MessageSource and has the messages in servlet-map/locale/messages.properties. Please use one or the other, but not both.